test/gui/shared/steps/steps.py

```diff
@@ -306,13 +306,17 @@ def step(context, fileName):
 @When('the user creates a file "|any|" with the following content on the file system')
 def step(context, filename):
     fileContent = "\n".join(context.multiLineText)
-    f = open(context.userData['clientSyncPathUser1'] + filename, "w")
+    f = open(context.userData['currentUserSyncPath'] + filename, "w")
     f.write(fileContent)
     f.close()
 
 
 @Given(r"^(.*) on the server (.*)$", regexp=True)
 def step(context, stepPart1, stepPart2):
+    isUserCreation = re.search('user "([^"]*)" has been created', stepPart1.strip())
+    if isUserCreation:
+        username = re.findall('"([^"]*)"', stepPart1)[0]
+        createUserSyncPath(context, username)
```

Comment:
We need to make sure that when we call the user creation step of middleware, we should use `@Given(r"^(.*) on the server (.*)$", regexp=True)` , i.e. the `on the server` should be in the middle. That seems to be our convention at the moment, and should keep it in mind for new scenarios that we add.

Reply:
any reason not to create the sync path on every scenario start?

Reply:
For now, sync path is created based on the user creation step (so that when the user is created, sync path will also be created).
We can also create sync paths for given users on every scenario start but we have to know prior which users will be created and added to client in that scenario (which is not an issue though).

Possible ways:
- create sync path when the user is created (current approach)
- create sync path on every scenario start
- create sync path at the time of adding user to client

Reply:
you are right, when the scenario starts we don't know the needed user name yet.
I like the third idea, I think that would look less hacky and be less fragile